activejob/lib/active_job/enqueue_after_transaction_commit.rb

```diff
@@ -28,11 +30,15 @@ module EnqueueAfterTransactionCommit # :nodoc:
         end
 
         if after_transaction
-          ActiveRecord.after_all_transactions_commit(&block)
+          self.successfully_enqueued = true
```

Comment:
Dunno if the `successfully_enqueued = true` is really justified. Maybe what we'd want is to return the job but with `successfully_enqueued = false`?